static-site/src/components/ListResources/Modal.tsx

```diff
@@ -123,22 +131,28 @@ const Title = styled.div`
   padding-bottom: 15px;
 `
 
-function _snapshotSummary(dates) {
+function _snapshotSummary(dates: string[]) {
   const d = [...dates].sort()
-  const [d1, d2] = [d[0], d.at(-1)].map((di) => new Date(di));
-  const days = (d2-d1)/1000/60/60/24;
+  if (d.length < 1) throw new Error("Missing dates.")
+
+  const d1 = new Date(d.at( 0)!).getTime();
```

Comment:
> Why is this bang needed? the preceding line ensures that there is at lease one element in d.

Compiler is not smart enough to figure this out from this kind of the conditional. Connecting `.length` to `.at()` is tough for a type system.

What it can easily figure out is null checks. So a cleaner way would be to check values and throw after the non-fallible array access:

```ts
const t0 = d.at(0)?.getTime()
if(isNil(t0)) { // careful to not exclude the legit value `0`
    throw ...
}
// `t0` is guaranteed to be `number` in this branch
```

I would go further. Finding first and last value is a common enough "algorithm" that I would introduce a utility function:

```ts
export function firstLastOrThrow<T>(a: T[]): [T, T] {
    // TODO: use .at() or lodash head() and tail() along with null checks 
}
```

This would make the code very pretty and safe, with all dirty details hidden:

```ts
const [t0, t1] = firstLastOrThrow(dates) // guaranteed to return a pair of numbers or to  throw
```

Reply:
Another thing is that this entire calculation is wrong. For example, not all years have 365 days. Date-time calculations is a very hard topic, and it is somewhat unreasonable to try and do on the spot. There are specialized high-quality libraries exist to do this correctly, e.g. [luxon](https://moment.github.io/luxon/#/?id=luxon) (continuation of `moment.js`). And then for human durations there are also small libs (e.g. [this](https://github.com/EvanHahn/HumanizeDuration.js))

Reply:
> Another thing is that this entire calculation is wrong.

(Let's break this out to a separate conversation if we want to continue discussing it -- I am aware how fraught date parsing is but I'll argue that the data this function returns is (purposefully) vague such that the overall function is correct.)